src/utils/circuit-to-svg.tsx

```diff
@@ -0,0 +1,317 @@
+                  THREE.MathUtils.degToRad(component.rotation.x ?? 0),
+                  THREE.MathUtils.degToRad(component.rotation.y ?? 0),
+                  THREE.MathUtils.degToRad(component.rotation.z ?? 0)
+                )
+              }
+              scene.add(mesh)
+            }
+          } catch (error) {
+            console.error('Failed to create footprint geometry:', error)
+          }
+        }
+      } catch (error) {
+        console.error('Failed to create component:', error)
+        
+        // Add fallback box for failed components (like MixedStlModel does)
+        const geometry = new THREE.BoxGeometry(0.5, 0.5, 0.5)
+        const material = new THREE.MeshStandardMaterial({
+          color: 0xff0000,
+          transparent: true,
+          opacity: 0.25
+        })
+        const mesh = new THREE.Mesh(geometry, material)
+        
+        if (component.position) {
+          mesh.position.set(
+            component.position.x ?? 0,
+            component.position.y ?? 0,
+            (component.position.z ?? 0) + 0.5
+          )
+        }
+        scene.add(mesh)
+      }
+    }
+
+    // Add board geometry after components
+    const boardGeom = createBoardGeomFromSoup(circuitJson)
+    if (boardGeom) {
+      for (const geom of boardGeom) {
+        try {
+          const geometry = createGeometryFromPolygons(geom.polygons)
+          
+          const material = new THREE.MeshStandardMaterial({
+            color: new THREE.Color(
+              geom.color[0],
+              geom.color[1],
+              geom.color[2]
+            ),
+            metalness: 0.1,
+            roughness: 0.8,
+            opacity: 0.9,  // Slightly more transparent
+            transparent: true,
+            side: THREE.DoubleSide
+          })
+
+          const mesh = new THREE.Mesh(geometry, material)
+          scene.add(mesh)
+        } catch (error) {
+          console.error('Failed to create board geometry:', error)
+        }
+      }
+    }
+
+    // Add grid (matching CadViewerContainer)
+    const gridHelper = new THREE.GridHelper(100, 100)
+    gridHelper.rotation.x = Math.PI / 2
+    scene.add(gridHelper)
+
+    // Center and scale scene
+    const box = new THREE.Box3().setFromObject(scene)
+    const center = box.getCenter(new THREE.Vector3())
+    const size = box.getSize(new THREE.Vector3())
+    
+    scene.position.sub(center)
+    
+    const maxDim = Math.max(size.x, size.y, size.z)
+    if (maxDim > 0) {
+      const scale = (1.0 - padding / 100) / maxDim
+      scene.scale.multiplyScalar(scale * 100)
+    }
+
+    // Render and return SVG
+    renderer.render(scene, camera)
+    
+    const svgString = new dom.window.XMLSerializer()
+      .serializeToString(renderer.domElement)
+      .replace(/xmlns="[^"]*"\s+xmlns="[^"]*"/, 'xmlns="http://www.w3.org/2000/svg"')
+
+    return svgString
+  } catch (error) {
+    console.error('Failed to generate SVG:', error)
+    throw error
```

Comment:
don't use a big try catch here (why bother having this at all, you just rethrow it?)